activerecord/test/cases/persistence_test.rb

```diff
@@ -504,6 +505,15 @@ def test_create
     assert_equal("New Topic", topic_reloaded.title)
   end
 
+  def test_create_prefetched_pk
+    ActiveRecord::Schema.define do
+      create_table :post_with_prefetched_pks, force: true do |t|
+    end
+    end
```

Comment:
The new model can point to the existing `posts` table by defining `self.table_name = "posts"` 
Usually there is no need to have a new table.

However, I wonder if in this we actually do want a new table with special handling for the `id` column. I assume we would want to drop the auto-increment property from our table since the value for the primary key will be pre-fetched. I assume this will represent a more realistic pk prefetching setup. However I have to admit I'm not sure how a real world db schema would look in this case.

But the schema definition needs to go to 
https://github.com/rails/rails/blob/main/activerecord/test/schema/schema.rb#L974 
```suggestion
```

Reply:
Good point about removing the explicit schema definition.

I agree with your point about the new table being a more realistic setup, but for now I will follow your suggestion and have the new model point to the existing posts table.